measure.rb

```diff
@@ -1872,10 +1877,18 @@ def self.add_hot_water_and_appliances(runner, model, building, weather, spaces)
           capacity_kbtuh = water_heating_system_values[:heating_capacity] / 1000.0
           oncycle_power = 0.0
           offcycle_power = 0.0
+          if has_desuperheater
+            if not @related_hvac_list.include? relatedhvac
+              @related_hvac_list << relatedhvac
+              coil_object = get_desuperheatercoil(@hvac_map, relatedhvac, sys_id)
+            else
+              fail "RelatedHVACSystem '#{relatedhvac}' for water heating system '#{sys_id}' is already attached to another water heating system."
+          end
+          end
           success = Waterheater.apply_tank(model, runner, space, to_beopt_fuel(fuel),
                                            capacity_kbtuh, tank_vol, ef, re, setpoint_temp,
                                            oncycle_power, offcycle_power, ec_adj,
-                                           @nbeds, @dhw_map, sys_id)
+                                           @nbeds, @dhw_map, sys_id, has_desuperheater, coil_object)
```

Comment:
We'll want to allow desuperheater for tankless water heater and HPWH too, right? I guess HPWH can't be implemented right now because it requires connecting to a stratified tank (?), but we should be able to add support for tankless? If so, let's add a test HPXML for it.